### PR TITLE
change quota design

### LINKED ISSUE #18540: User management view: Rethink quota
**Is your feature request related to a problem? Please describe.**

Screenshot from 2019-12-22 19-56-22

I like the new user management view :+1: but I'm unhappy with the quota bar. Please switch from a designer to a user manager perspective for a moment. I'm not able to see what quota is assigned for a user or how many storage is used. 

- To see the storage in use (to concrete value): hover over the bar.
- To see the assigned quota: enter the edit mode.

**Describe the solution you'd like**
Remove the progress bar. Show the important information: `x MB of y MB used`. The progress bar is fine for the user view but not user management view.

**Describe alternatives you've considered**
There is no alternative to consider.

cc @nextcloud/designers

### PR DESCRIPTION
**Pull request by GretaD**
fixes #18540, #19969 
userquota

### CODE REVIEW

**Review by jancborchardt — DISMISSED** [review]
- [x] The text should be left-aligned, as with the other entries
- [ ] If nothing is used, it can be "(0 used)", no unit needed
- [ ] Would still be nice to get the bar back, otherwise people need to read it all and calc in their head

Otherwise it looks good! :)

**Comment by kesselb** [comment]
Thanks @GretaD :+1: 

image

```
Index: apps/settings/src/components/UserList/UserRowSimple.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/settings/src/components/UserList/UserRowSimple.vue	(revision 8d84710e266bd821fec441b238f7801216c8d11a)
+++ apps/settings/src/components/UserList/UserRowSimple.vue	(date 1579182181172)
@@ -33,7 +33,12 @@
 			{{ userSubAdminsGroupsLabels }}
 		</div>
 		<div class="quota">
-			{{ userQuota }} ({{ usedSpace }})
+			<span>{{ userQuota }} ({{ usedSpace }})</span>
+			<progress
+					class="quota-user-progress"
+					:class="{'warn': usedSpace > 80}"
+					:value="usedSpace"
+					max="100" />
 		</div>
 		<div v-if="showConfig.showLanguages" class="languages">
 			{{ userLanguage.name }}
@@ -169,4 +174,13 @@
 		text-overflow: ellipsis;
 		white-space: nowrap;
 	}
+	.quota span {
+		margin-left: 10px;
+	}
+	.quota progress {
+		position: absolute;
+		bottom: 0;
+	}
 </style>
```

I guess there are cleaner ways to do it ;) Also some tweaks for smaller screens would be good.

File: apps/settings/css/settings.scss [hunk 1]
```diff
@@ -1680,6 +1680,7 @@ doesnotexist:-o-prefocus, .strengthify-wrapper {
 
 				&.quota {
 					height: 44px;
+					min-width: 150px;
```
Comment (juliushaertl) [inline apps/settings/css/settings.scss]:
```suggestion
					min-width: $grid-col-min-width;
```

**Comment by kesselb** [comment]
Needs a rebase.

**Comment by kesselb** [comment]
image

Looks like that on my machine. I run make all a few times but the result seems to be the same.

- [ ] The quota header column should be aligned with the other header columns (text left not centered)
- [ ] Also the quota text for a user is centered
- [ ] The overlapping with the edit button is probably fixed with a rebase

File: apps/settings/src/components/UserList/UserRowSimple.vue [hunk 2]
```diff
@@ -145,6 +146,15 @@ export default {
 		canEdit() {
 			return getCurrentUser().uid !== this.user.id && this.user.id !== 'admin'
 		},
+		userQuota() {
+			if (this.user.quota.quota === 'none') {
+				return t('settings', 'Unlimited')
+			}
+			if (this.user.quota.quota >= 0) {
+				return t('settings', '{size}', { size: OC.Util.humanFileSize(this.user.quota.quota) })
```
Comment (juliushaertl) [inline apps/settings/src/components/UserList/UserRowSimple.vue]: No need to translate this :wink:

File: apps/settings/src/components/UserList/UserRowSimple.vue [hunk 3]
```diff
@@ -145,6 +146,15 @@ export default {
...
+			}
+			return t('settings', '{size}', { size: OC.Util.humanFileSize(0) })
```
Comment (juliushaertl) [inline apps/settings/src/components/UserList/UserRowSimple.vue]: Same here ;)

**Comment by kesselb** [comment]
image

:confused:

**Comment by GretaD** [comment]
> image
> 
> confused

ouch, what happened there? /hides

**Comment by kesselb** [comment]
image

Looks good to me.

image

Would be nice to see the current quota also in edit mode. But not a blocker for this pr.

**Review by kesselb — APPROVED** [review]

**Comment by juliushaertl** [comment]
The quota bar positioning still seems a bit off 
image

File: apps/settings/css/settings.scss [hunk 4]
```diff
@@ -1610,15 +1610,16 @@ doesnotexist:-o-prefocus, .strengthify-wrapper {
...
+					white-space: nowrap;
+					position: relative;
 
 					progress {
-						width: 100%;
-						margin: 0 10px;
+						width: 150px;
+						margin-top: 35px;
 						height: 3px;
+						position: absolute;
```
Comment (juliushaertl) [inline apps/settings/css/settings.scss]: I would suggest to wrap both the text and process in a wrapping div so you can more easily style those without the need of a absolute positioning.

**Comment by kesselb** [comment]
cc @GretaD mind to rebase?

**Comment by tanderson8** [comment]
apps/settings/src/components/UserList/UserRowSimple.vue

 I'm not seeing an /src directory from my /settings directory. Do I need to just add it and push the file to it for these changes to take effect?

**Comment by GretaD** [comment]
@tanderson8 is you master up to date? You normally should have the UserRowSimple.vue file when you checkout out at rethink_quota branch.

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
Let's do this. The quota/language assignment issue will be fixed by https://github.com/nextcloud/nextcloud-vue/pull/924

@GretaD Mind to rebase and commit the bundles?

**Comment by kesselb** [comment]
This pull request reverts https://github.com/nextcloud/server/pull/18555 partly. It's again not possible to edit my own user account. 

cc @GretaD @juliushaertl

File: apps/settings/src/components/UserList/UserRowSimple.vue [hunk 5]
```diff
@@ -143,9 +147,19 @@ export default {
 			}
 			return t('settings', '{size} used', { size: OC.Util.humanFileSize(0) })
 		},
-		canEditUser() {
```
Comment (juliushaertl) [inline apps/settings/src/components/UserList/UserRowSimple.vue:146]: @GretaD It seems this was reverted for some reason as @kesselb mentioned as well as the related part in the template. I did not notice that during review, so it was probably during rebase? Could you look into fixing that again?

Comment (GretaD) [reply apps/settings/src/components/UserList/UserRowSimple.vue:146]: yes sure

**Pull request merged**